### DISCUSSION TITLE
Copy and Paste ?

### DISCUSSION THREAD

**Discussion by renueman** [Ylianst/MeshCentral#4919] [General]
I have noticed that 'copy and paste' is not working on any of my clients. I have tried this on my local lan and to clients across the net. When I copy text from my system and then go over to the remote client paste is greyed out. Perhaps I am doing something wrong or I have a setting not turned on?

I connect to both windows and linux clients.

Nothing is more frustrating than having to type in long strings on the remote system. I was not going to post about this here because I figured its going to be something stupid and I would look like a fool. Well, I guess now I will take that chance :(

Any help would be appreciated.

**Comment by bbrendon** [comment]
There is a button in the tool bar to transfer the clipboard from remote to local and vise versa. You have to use that before the clipboard will populate with your data.

**Reply by renueman** [reply]
Playing around with it here to see how it works. I am logged into my Debian Server with the text editor open. I type in something, Highlight it all with my mouse and then choose copy. Then I clicked on 'clipboard' and then 'get clipboard' and it worked. Now, I have not figured out the steps to paste from my local computer to the remote one ? . I have no idea what 'Set Clipboard' does. Is there a doc on this somewhere that I can read up on this?

Thank you for your reply.

**Reply by lexisman** [reply]
> There is a button in the tool bar to transfer the clipboard from remote to local and vise versa. You have to use that before the clipboard will populate with your data.

It used to have 2 buttons, one for uploading and the other for downloading, however the uploading is not there anymore on the latest version, 1.1.2. However there is a button on the left hand side that is labelled "clipboard", it can be used to upload whatever is in the local clipboard to the remote. When you click the button "clipboard" then paste whatever is in the local clipboard, then click "set clipboard". Now you have the local clipboard copied to the remote clipboard.

**Reply by renueman** [reply]
@lexisman Thank you very much. I followed your directions doing a cp/paste from my linux mint to centos7 remote server. I opened up a GUI text editor on the server but right clicking on the text editor the paste option was greyed out. So, I reverted back to cntrl-V and it worked. 
My life just got easier ;)

**Comment by silversword411** [comment]
You looked at the docs? https://ylianst.github.io/MeshCentral/meshcentral/devicetabs/

**Reply by renueman** [reply]
Yes, I noticed that the icons on the lower right of my screen all appear to be greyed out. However, That just might be the 'look' for them. I also noticed there is only one clipboard icon - mousing over it says "Download clipboard from remote". In the docs it has a picture of two. One for 'remote to local' and one for 'local to remote'. I am missing the 'local to remote' icon. Hummm... the plot thickens...

**Reply by silversword411** [reply]
There's browser security that comes into play with clipboard (Firefox has a permission for auto-clipboard I believe).

If you find solutions doing a PR to the docs for the next person would be a nice thing to do 🙂

**Reply by renueman** [reply]
`If you find solutions doing a PR to the docs for the next person would be a nice thing to do`

PR to the docs ?

**Reply by renueman** [reply]
Pull Request! Pull Request!! ---- Ok, Got it.. lol...
